sanic/signals.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+import asyncio
+
+from inspect import isawaitable
+from typing import Any, Dict, Optional
+
+from sanic_routing import BaseRouter, Route  # type: ignore
+from sanic_routing.utils import path_to_parts  # type: ignore
+
+from sanic.exceptions import InvalidSignal, SanicException
+from sanic.models.handler_types import SignalHandler
+
+
+class Signal(Route):
+    def get_handler(self, raw_path, method, _):
+        method = method or self.router.DEFAULT_METHOD
+        raw_path = raw_path.lstrip(self.router.delimiter)
+        try:
+            return self.handlers[raw_path][method]
+        except (IndexError, KeyError):
+            raise self.router.method_handler_exception(
+                f"Method '{method}' not found on {self}",
+                method=method,
+                allowed_methods=set(self.methods[raw_path]),
+            )
+
+
+class SignalRouter(BaseRouter):
+    def __init__(self) -> None:
+        super().__init__(
+            delimiter=".",
+            route_class=Signal,
+            stacking=True,
+        )
+        self.ctx.loop = None
+
+    def get(  # type: ignore
+        self,
+        event: str,
+        extra: Optional[Dict[str, str]] = None,
+    ):
+        extra = extra or {}
+        return self.resolve(f".{event}", extra=extra)
+
+    async def _dispatch(
+        self,
+        event: str,
+        *fields,
+        context: Optional[Dict[str, Any]] = None,
+        where: Optional[Dict[str, str]] = None,
+    ) -> None:
+        if fields:
+            try:
+                event = self.delimiter.join([event, *fields])
```

Comment:
Is there any advantages to supporting this ?  Can we not just enforce you send the full even as a single string for the first argument instead ? It will also let us avoid and entire check